### PR TITLE
Chore: Small cleanups for clarity

### PR DESCRIPTION
**Pull request by jannotti**
I'm doing a larger bit of work to enable logicsig cost pooling and I found myself making these changes along the way. It's easier to review by extracting them into their own PR.  This does four things.

1. Introduces and uses:
```
func (msig MultisigSig) Signatures() int
```
for the several places we need to examine a MultiSig to count the number of sub signatures present.

2. Uses the pre-existing `func (s *Signature) Blank() bool` in many places.

3. Renames `EvalContext.runModeFlags` to simply `EvalContext.runMode` because it takes on a single value, it can't be, for example: `ModeSig | ModeApp`, it must be one or the other.

4. Improves tests by making it so that sample transactions created for versions below `rekeyingEnabledVerion` don't have their `RekeyAddress` set.

### CODE REVIEW

**Review by ahangsu — DISMISSED** [review]
looks good

File: crypto/multisig.go [hunk 1]
```diff
@@ -75,6 +75,18 @@ func (msig MultisigSig) Preimage() (version, threshold uint8, pks []PublicKey) {
 	return msig.Version, msig.Threshold, pks
 }
 
+// Signatures returns the actually number of signatures included in the
```
Comment (jasonpaulos) [inline crypto/multisig.go:78]:
```suggestion
// Signatures returns the actual number of signatures included in the
```
nit

**Review by jasonpaulos — APPROVED** [review]
Good improvements

File: crypto/multisig.go [hunk 2]
```diff
@@ -249,37 +261,18 @@ func MultisigBatchPrep(msg Hashable, addr Digest, sig MultisigSig, batchVerifier
...
-
 	// checks the number of non-blank signatures is no less than threshold
-	var counter uint8
-	for _, subsigi := range sig.Subsigs {
-		if (subsigi.Sig != Signature{}) {
-			counter++
-		}
-	}
-	if counter < sig.Threshold {
+	if sig.Signatures() < int(sig.Threshold) {
```
Comment (algorandskiy) [inline crypto/multisig.go:265]: this appears to be a protocol change - old and new version process blank multisig txns differently

Comment (jannotti) [reply crypto/multisig.go:265]: What is the difference? `sig.Signatures()` does the same thing as the loop, so this line do the same thing as it was before. If there are no signatures (and sig.Threshold is > 0) then `errInvalidNumberOfSignature` is returned.

Comment (algorandskiy) [reply crypto/multisig.go:265]: I was thought about blank signatures in this loop above:
```
	if len(sig.Subsigs) < int(sig.Threshold) {
		return errInvalidNumberOfSignature
	}
```
Now I see, this is only a preliminary check and the new and old code are functionally equivalent, sorry.

Comment (jannotti) [reply crypto/multisig.go:265]: No problem, I always appreciate a double-check.

File: crypto/multisig.go (hunk 1 above)
Comment (jannotti) [reply crypto/multisig.go]: fixed

**Review by ahangsu — APPROVED** [review]

**Review by jasonpaulos — APPROVED** [review]

**Review by ASISBusiness — COMMENTED** [review]
Gn

**Pull request merged**